app/javascript/mastodon/components/edited_timestamp/index.js

```diff
@@ -56,12 +55,12 @@ class EditedTimestamp extends React.PureComponent {
   }
 
   render () {
-    const { timestamp, intl, statusId } = this.props;
+    const { timestamp, statusId } = this.props;
 
     return (
       <DropdownMenu statusId={statusId} renderItem={this.renderItem} scrollable renderHeader={this.renderHeader} onItemClick={this.handleItemClick}>
         <button className='dropdown-menu__text-button'>
-          <FormattedMessage id='status.edited' defaultMessage='Edited {date}' values={{ date: intl.formatDate(timestamp, { hour12: false, month: 'short', day: '2-digit', hour: '2-digit', minute: '2-digit' }) }} /> <Icon id='caret-down' />
+          <FormattedMessage id='status.edited' defaultMessage='Edited {date}' values={{ date: new Date(timestamp).toLocaleString(undefined, { hourCycle: 'h23', month: 'short', day: '2-digit', hour: '2-digit', minute: '2-digit' }) }} /> <Icon id='caret-down' />
```

Comment:
`undefined` would imply that navigator locale should be used, which would look inconsistent if user's Mastodon language does not match browser language. Instead you probably should keep the `intl` injecton and use `intl.locale` to retrieve currently used locale by Mastodon.

Reply:
On principle, I'm disinclined to include intl for that singular purpose.

I'm honestly not sure how I feel about this one: as a user I could see it going either way; as far as I'm aware, the most common use-case for changing Mastodon locale to something other than browser locale is for native language use in a foreign country where you want the browser to report your physical locale while having your social media displayed in your native format.

In that scenario, having the time format match your browser locale seems appropriate? Since that would be consistent with every other presentation of datetime in the browser, which presumably is aligned with daily life for the user. Time display and language defaults aren't necessarily aligned.

Thoughts?

Reply:
I don't think you need to consider reasons why users change languages on sites, in browsers or system-wide, because there a lot of them, none less valid than the other. And from personal experience, some of them can be quite ridiculous, for example, I sometimes change language just to make a screenshot in that language, in which case changing the language on the site is much quicker than changing it in browser or system. For translator, it's also helpful to test translations in context.

If site displays something that does not match my preferences, it looks broken to me. Imagine GitHub, fully Englisch site, started displaying time indicators in your browser's language, this would look very inconsistent (‘Edited 1 гадзіну таму’). It's not that if you don't provide language to use the browser will magically match the formatting preferences in system, they don't do that, neither any allow customisability.

It's absolutely not worth sacrificing consistency for no good reason, it also can be harmful because language of formatted dates then won't match the language that Mastodon tells it uses (`lang` attribute on `html` element), posing problems for users with screen readers. Those are my thoughts.

Reply:
All absolutely valid thoughts and I concede the point :)

Is there no way to obtain the locale from Mastodon without using `intl` and pass that directly to the browser API instead of undefined? I'd rather not have the dependency on a library that we know has problems.

Reply:
In `@formatjs/intl`, the library that powers `react-intl`, `locale` property is taken directly from the configuration and never modified. The `locale` in configuration is defined here: 

https://github.com/mastodon/mastodon/blob/ef4d29c8791086b11f6e36aa121ff5c9b5fa0103/app/javascript/mastodon/containers/mastodon.js#L82

As you can see, it is defined using `getLocale` from  `mastodon/locales`, therefore using this property alone should not cause any issues unless Mastodon itself was to use wrong language codes.

You could of course use this `getLocale` yourself, but I absolutely would advise against doing that, since it creates hard dependency and would make it more difficult for Mastodon to add language changing without hard reload in the future (which I believe is in the plans because of the unifying UI goal); by using injection, whenever any of `IntlProvider` properties change, it would update the injection in all of children components, causing components to update too.

Reply:
Oh I see. I misread. I have no objections to using `intl` and was confusing it with `react-intl`

Thanks for the dialog! I'll get this changed

Reply:
I've added the fix for this into the PR - they should now all use the locale from the config rather than the browser's locale